classes/class-gf-securesubmit.php

```diff
@@ -1481,36 +1493,17 @@ private function authorizeCC($feed, $submission_data, $form, $entry)
             $currency = GFCommon::get_currency();
             $transaction = null;
             if ($isAuth) {
-                $transaction = $service->authorize(
-                    $submission_data['payment_amount'],
-                    $currency,
-                    $token,
-                    $cardHolder,
-                    false,
-                    null,
-                    null,
-                    false,
-                    $cpcReq,
-                    null,
-                    null,
-                    $secureEcommerce
-                );
+                $transaction = $cardHolder->authorize($submission_data['payment_amount'])
+                                ->withCurrency($currency)
+                                ->withAddress($address)
+                                ->withAllowDuplicates(true)
+                                ->execute();
             } else {
-                $transaction = $service->charge(
-                    $submission_data['payment_amount'],
-                    $currency,
-                    $token,
-                    $cardHolder,
-                    false,
-                    null,
-                    null,
-                    false,
-                    $cpcReq,
-                    null,
-                    null,
-                    null,
-                    $secureEcommerce
-                );
+                $transaction = $cardHolder->charge($submission_data['payment_amount'])
+                                ->withCurrency($currency)
+                                ->withAddress($address)
+                                ->withAllowDuplicates(true)
+                                ->execute();       
```

Comment:
The data within `$secureEcommerce` and the level II / corporate purchasing card flag (`$cpcReq`) are no long included in the original authorization request.

Reply:
updated

Reply:
@pkuma525 I may not have conveyed the right meaning here. The `$secureEcommerce` and `$cpcReq` values can still be used in the new SDK, but your original changes did not add these values to the authorization builder, causing them to not be included in the request to the gateway. We need both to be included in the gateway request to leverage both programs.

Reply:
updated as per review comment